File: objectiveai-rs/src/functions/executions/request/reasoning.rs
//! Reasoning configuration for function executions.

use crate::agent;
use serde::{Deserialize, Serialize};
use schemars::JsonSchema;

/// Configuration for generating reasoning summaries during execution.
///
/// When enabled, an LLM summarizes the execution's reasoning process.
#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
#[schemars(rename = "FunctionsExecutionsRequestReasoning")]
pub struct Reasoning {
    /// The primary agent to use for generating reasoning summaries.
    pub agent: agent::completions::request::Agent,
    /// Fallback agents tried in order if the primary is rate-limited or errors.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub agents: Option<Vec<agent::completions::request::Agent>>,
}

File: objectiveai-js/src/functions/executions/request/reasoning.ts
import { z } from "zod";
import { AgentCompletionsRequestAgentSchema } from "../../../agent/completions/request/agent";

export const FunctionsExecutionsRequestReasoningSchema = z.object({
  agent: AgentCompletionsRequestAgentSchema.describe("The primary agent to use for generating reasoning summaries."),
  agents: z.array(AgentCompletionsRequestAgentSchema).nullable().describe("Fallback agents tried in order if the primary is rate-limited or errors.").optional(),
}).describe("Configuration for generating reasoning summaries during execution.\n\nWhen enabled, an LLM summarizes the execution's reasoning process.").meta({ title: "functions.executions.request.Reasoning" });
export type FunctionsExecutionsRequestReasoning = z.infer<typeof FunctionsExecutionsRequestReasoningSchema>;
